### PR TITLE
Handle exported functions in ExportDeclarations for CJS/AMD/UMD emit

### LINKED ISSUE #58473: 5.5.0 regression - Named exports of functions are not emitted for CommonJS
### 🔎 Search Terms

commonjs export

### 🕗 Version & Regression Information

- This changed between versions 5.4.5 and 5.5.0

### ⏯ Playground Link

https://www.typescriptlang.org/play/?target=7&module=1&isolatedModules=true&verbatimModuleSyntax=false&ts=5.5.0-beta#code/GYVwdgxgLglg9mABMOcAUBKAXIgbnGAE0QG8AoAXzLIFMAPABzgCcpSzFPlUAaSgbjJA

### 💻 Code

```ts
function foo(): void {
}

export {
    foo,
};
```

### 🙁 Actual behavior

ts 5.5.0-beta and 5.5.0-dev-20240508 both emit:

```js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.foo = void 0;
function foo() {
}
```

note missing last line

### 🙂 Expected behavior

ts 5.4.5 emit

```js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.foo = void 0;
function foo() {
}
exports.foo = foo;

```

### Additional information about the issue

_No response_

### PR DESCRIPTION
**Pull request by jakebailey**
Fixes #58473

Not 100% certain on this but the baselines look right. Feel free to make your own better fix for this, though; this was just my naive attempt.

### CODE REVIEW

File: tests/baselines/reference/jsDeclarationsDefault.js [hunk 1]
```diff
@@ -46,7 +46,9 @@ exports.default = 12;
 //// [index2.js]
 "use strict";
 Object.defineProperty(exports, "__esModule", { value: true });
-exports.bar = exports.x = void 0;
+exports.x = void 0;
+exports.default = foo;
+exports.bar = foo;
 exports.default = foo;
 exports.bar = foo;
```
Comment (jakebailey) [inline tests/baselines/reference/jsDeclarationsDefault.js:53]: This duplication doesn't feel too great. I probably missed something.

File: tests/baselines/reference/jsDeclarationsFunctions.js [hunk 2]
```diff
@@ -63,15 +63,20 @@ export function j() {}
...
 Object.defineProperty(exports, "__esModule", { value: true });
-exports.jj = exports.ii = exports.h = exports.g = void 0;
 exports.a = a;
 exports.b = b;
 exports.c = c;
 exports.d = d;
 exports.e = e;
 exports.f = f;
+exports.g = g;
+exports.h = hh;
```
Comment (jakebailey) [inline tests/baselines/reference/jsDeclarationsFunctions.js:73]: Example of this working, in the test case:

```js
/**
 * @param {{x: string}} a
 * @param {{y: typeof b}} b
 */
function hh(a, b) {
    return a.x && b.y();
}
export { hh as h };
```

File: tests/baselines/reference/jsDeclarationsDefault.js (hunk 1 above)
Comment (jakebailey) [reply tests/baselines/reference/jsDeclarationsDefault.js]: Fixed this via a Set.

**Comment by jakebailey** [comment]
@typescript-bot perf test this faster

**Review by rbuckton — APPROVED** [review]

**Pull request merged**